### ISSUE TITLE
Read file with delimiter

### ISSUE THREAD

**Issue by 2005m** [vlang/v#13905]
Hi,
The following post explains how to read a csv file:
https://stackoverflow.com/questions/64726760/reading-all-csv-lines
Is there an easy way to read a delimited file? i.e. same as csv but instead of `,` using a different delimiter? I was able to change the source code of reader.v to do it. I was wondering if there is an easier way without breaking change to the source code?
I was also thinking that the example on stackoverflow could be added to your examples folder.
Thank you

**Comment by SurmanPP** [comment]
The delimeter is defined by a struct member and the `new_reader()` function initlaizes this member as ',' (comma). Instead of using that function you can create a Reader struct "manually" like:
```V
mut my_reader = rReader{
delimeter: '.'
comment: '#'
data: my_string_to_read
}
```
And then you can use the normal reader functions with this struct.

# TL;DR
Create the struct manually instead of using the function `new_reader()`

**Comment by 2005m** [comment]
I have tried the following but it does not seem to work:
```
import os
import encoding.csv 

struct Reader { 
  data string
pub mut:	
delimiter byte	
comment byte
}

fn my_reader(data string, delimiter byte) &Reader {	return &Reader{
delimiter: delimiter
comment: `#`
data: data
}
}

fn main() { 
content := os.read_file('file.txt') ?
mut reader := my_reader(content, `.`)
for {
 line_data := reader.read() or {
 break 
} 
println(line_data)
 } 
}
```
But I get errors when compiling:
1. Warning that module csv is importes but never used
2. Error unknown method or field Reader.read
3. Error println can not print void

Seems like using the normal reader function doea not work.
Thank you

**Comment by StunxFS** [comment]
@2005m that code is wrong, it should use `csv.Reader`, but this struct is private.

It would be a matter of modifying that module and providing a function that allows using a custom delimiter.

Something like that:

```v
pub fn new_reader_with_custom_delimiter(data string, delimiter byte) &Reader {
	return &Reader{
		delimiter: delimiter
		comment: `#`
		data: data
	}
}
```

**Comment by StunxFS** [comment]
It would also be valid to use a struct marked as `[params]`:

```v
[params]
struct ReaderConfig {
	data string
	delimiter byte = `,`
	comment byte = `#`
}

pub fn new_reader(config ReaderConfig) &Reader {
	return &Reader{
		delimiter: config.delimiter
		comment: config.comment
		data: config.data
	}
}

x := new_reader(data: "Text", delimiter: `|`)
```

**Comment by 2005m** [comment]
Thank you all. That was helpful.
